scripts-components/in-view/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ComponentInitializationError } from '@mangoweb/scripts-base'
+import { IsInView, isInView, isOutOfView, isSupported } from 'isinview'
+
+interface InViewData {
+	targets?: string
+	threshold?: number
+
+	isAboveViewClass?: string
+	isInViewClass?: string
+	isBelowViewClass?: string
+	isSeenClass?: string
+}
+
+export class InView extends Component<InViewData> {
+	static componentName = 'InView'
+
+	protected readonly targets: IsInView.Target
+	protected readonly options: Partial<IsInView.Options> = {}
+
+	protected readonly isAboveViewClass: string
+	protected readonly isInViewClass: string
+	protected readonly isBelowViewClass: string
+	protected readonly isSeenClass: string
+
+	public constructor(el: HTMLElement, data: InViewData) {
+		super(el, data)
+
+		if (!isSupported()) {
+			throw new ComponentInitializationError('InView component not supported. Add IntersectionObserver polyfill.')
+		}
+
+		this.targets = this.data.targets ? this.getChildren(this.data.targets) : this.el
+
+		if (this.data.threshold) {
+			this.options.threshold = this.data.threshold
+		}
+
+		this.isAboveViewClass = this.getProp('isAboveViewClass', 'is-aboveView')
+		this.isInViewClass = this.getProp('isInViewClass', 'is-inView')
+		this.isBelowViewClass = this.getProp('isBelowViewClass', 'is-belowView')
+		this.isSeenClass = this.getProp('isSeenClass', 'is-seen')
+	}
+
+	public init() {
+		isInView(
+			this.targets,
+			(target, details) => {
+				target.classList.add(this.isSeenClass)
+				this.togglePositionClasses(target, details)
+			},
+			this.options
+		)
+
+		isOutOfView(
+			this.targets,
+			(target, details) => {
```

Comment:
Když si z `togglePositionClasses` uděláš property, můžeš tady psát
```ts
isOutOfView(
	this.targets,
	this.togglePositionClasses,
	this.options
)
```

Reply:
Díky za tip. Upravil jsem to podle tvého návrhu.